src/waveform/renderers/allshader/waveformrendermark.cpp

```diff
@@ -383,20 +388,22 @@ void allshader::WaveformRenderMark::drawUntilMark(const QMatrix4x4& matrix, floa
 // Note that in the legacy waveform widgets this is drawn directly
 // in the WaveformWidgetRenderer itself. Doing it here is cleaner.
 void allshader::WaveformRenderMark::updatePlayPosMarkTexture() {
-    float imgwidth;
-    float imgheight;
+    const float imgheight = m_waveformRenderer->getBreadth();
+    const float imgwidth = 11.f;
 
-    const float height = m_waveformRenderer->getBreadth();
-    const float devicePixelRatio = m_waveformRenderer->getDevicePixelRatio();
+    if (imgheight == 0.0f) {
+        return;
+    }
 
+    const float devicePixelRatio = m_waveformRenderer->getDevicePixelRatio();
     const float lineX = 5.5f;
 
-    imgwidth = 11.f;
-    imgheight = height;
-
     QImage image(static_cast<int>(imgwidth * devicePixelRatio),
             static_cast<int>(imgheight * devicePixelRatio),
             QImage::Format_ARGB32_Premultiplied);
+    if (image.isNull()) {
+        return;
+    }
```

Comment:
imgWidth is < 0 and there is already a check for `imgHeight != 0` above, so this can only be NULL if devicePixelRatio is 0.
Can this ever happen?

Reply:
When looking at the Qt source, there are other reasons like out of memory or such. All can't probably not happen but this saves Mixxx from crash if it happen anyway. I think a 
VERIFY_OR_DEBUG_ASSERT is a good choice.